src/robusta/integrations/slack/sender.py

```diff
@@ -645,6 +765,99 @@ def send_finding_to_slack(
             thread_ts=thread_ts,
         )
 
+    def __send_finding_to_slack_preview(
+        self,
+        finding: Finding,
+        sink_params: SlackSinkPreviewParams,
+        platform_enabled: bool,
+        thread_ts: str = None,
+    ) -> str:
+        blocks: List[SlackBlock] = []
+        attachment_blocks: List[BaseBlock] = []
+
+        slack_channel = ChannelTransformer.template(
+            sink_params.channel_override,
+            sink_params.slack_channel,
+            self.cluster_name,
+            finding.subject.labels,
+            finding.subject.annotations,
+        )
+
+        if finding.finding_type == FindingType.AI_ANALYSIS:
+            # holmes analysis message needs special handling
+            self.send_holmes_analysis(finding, slack_channel, platform_enabled, thread_ts)
+            return ""  # [arik] Looks like the return value here is not used, needs to be removed
+
+        status: FindingStatus = (
+            FindingStatus.RESOLVED if finding.title.startswith("[RESOLVED]") else FindingStatus.FIRING
+        )
+
+        if finding.title:
+            blocks.extend(self.__create_finding_header_preview(finding, status, platform_enabled,
+                                                         sink_params.investigate_link, sink_params))
+
+        if not sink_params.hide_buttons:
+            links_block: LinksBlock = self.__create_links(
+                finding, platform_enabled, sink_params.investigate_link, sink_params.prefer_redirect_to_platform
+            )
+            blocks.append(links_block)
+
+            if HOLMES_ENABLED and HOLMES_ASK_SLACK_BUTTON_ENABLED:
+                blocks.append(self.__create_holmes_callback(finding))
+
+        if not sink_params.get_custom_template():
+            blocks.append(MarkdownBlock(text=f"*Source:* `{self.cluster_name}`"))
+
+        if not sink_params.get_custom_template() and finding.description:
+            if finding.source == FindingSource.PROMETHEUS:
+                blocks.append(MarkdownBlock(f"{Emojis.Alert.value} *Alert:* {finding.description}"))
+            elif finding.source == FindingSource.KUBERNETES_API_SERVER:
+                blocks.append(
+                    MarkdownBlock(f"{Emojis.K8Notification.value} *K8s event detected:* {finding.description}")
+                )
+            else:
+                blocks.append(MarkdownBlock(f"{Emojis.K8Notification.value} *Notification:* {finding.description}"))
+        unfurl = True
+
+        if sink_params.hide_enrichments:
```

Comment:
why not just adding the enrichment blocks under `if not sink_params.hide_enrichments:`